subsys/mpsl/fem/common/mpsl_fem_utils.c

```diff
@@ -39,20 +39,20 @@ void mpsl_fem_pin_extend_with_port(uint8_t *pin, const char *lbl)
 	 */
 
 #if DT_NODE_HAS_STATUS(DT_NODELABEL(gpio0), okay)
-	if (strcmp(lbl, DT_LABEL(DT_NODELABEL(gpio0))) == 0) {
+	if (strcmp(lbl, UTIL_CAT(DT_NODELABEL(gpio0), _FULL_NAME)) == 0) {
 		return;
 	}
 #endif
 
 #if DT_NODE_HAS_STATUS(DT_NODELABEL(gpio1), okay)
-	if (strcmp(lbl, DT_LABEL(DT_NODELABEL(gpio1))) == 0) {
+	if (strcmp(lbl, UTIL_CAT(DT_NODELABEL(gpio1), _FULL_NAME)) == 0) {
 		*pin += 32;
 		return;
 	}
```

Comment:
why isn't this fixed by removing usage of labels?

Reply:
There needs to be a way of knowing if the GPIO is part of port 0 or 1 so that the "combined" pin number (e.g. p0.03 = 3 and p1.03 = 32 + 3 = 35) can be used for direct register access. I don't like this code, it shouldn't be doing a string compare at all, but I'm not aware of another way of knowing if a pin belongs to port 0 or 1. This doesn't check the label, this actually checks the full name of the dts node.

Reply:
This code assumes that pins numbered `0..31` belong to port 0 and `32..63` to port 1. Since these string comparisons have become quite cumbersome, maybe it would be enough to take full advantage of the above assumption and implement it as:

```c
void mpsl_fem_pin_extend_with_port(uint8_t *pin)
{
	/* The pin numbering in the FEM configuration API follows the
	 * convention:
	 *   pin numbers 0..31 correspond to the gpio0 port
	 *   pin numbers 32..63 correspond to the gpio1 port
	 *
	 * Values obtained from devicetree are here adjusted to the ranges
	 * given above.
	 */
#if DT_NODE_HAS_STATUS(DT_NODELABEL(gpio0), okay)
	if (*pin < 32) {
		return;
	}
#endif
#if DT_NODE_HAS_STATUS(DT_NODELABEL(gpio1), okay)
	if (*pin >= 32) {
		*pin += 32;
		return;
	}
#endif
	(void)pin;

	__ASSERT(false, "Unknown GPIO port");
}
```

Reply:
The dts mapping does take into consideration the pin number for the port, e.g.:
```
        nrf_radio_fem: nrf21540_fem {
                compatible = "nordic,nrf21540-fem";
                tx-en-gpios = <&arduino_header 11 GPIO_ACTIVE_HIGH>;   /* D5 */
                rx-en-gpios = <&arduino_header 9 GPIO_ACTIVE_HIGH>;    /* D3 */
                pdn-gpios = <&arduino_header 15 GPIO_ACTIVE_HIGH>;     /* D9 */
                ant-sel-gpios = <&arduino_header 10 GPIO_ACTIVE_HIGH>; /* D4 */
                mode-gpios = <&arduino_header 8 GPIO_ACTIVE_HIGH>;     /* D2 */
                spi-if = <&nrf_radio_fem_spi>;
        };
```
and
```
        nrf_radio_fem: nrf21540_fem {
                compatible = "nordic,nrf21540-fem";
                tx-en-gpios = <&gpio0 22 GPIO_ACTIVE_HIGH>;
                rx-en-gpios = <&gpio0 19 GPIO_ACTIVE_HIGH>;
                pdn-gpios = <&gpio0 23 GPIO_ACTIVE_HIGH>;
                ant-sel-gpios = <&gpio0 20 GPIO_ACTIVE_HIGH>;
                mode-gpios = <&gpio0 17 GPIO_ACTIVE_HIGH>;
                spi-if = <&nrf_radio_fem_spi>;
        };
```
So the pin number going into this function is between 0-31, however it needs to be the old style pin number coming out, 0-47, and the only way to do that is to check if the incoming GPIO node is GPIO0 or GPIO1

Reply:
Oops, sure. My proposal doesn't make any sense. Sorry for the noise

Reply:
There's the `NRF_DT_GPIOS_TO_PSEL` helper (defined in `soc_nrf_common.h`).

Reply:
Thanks @gmarull, switched over to device tree macros!